homeassistant/components/caldav/calendar.py

```diff
@@ -142,8 +142,22 @@ def __init__(self, calendar, include_all_day, search):
             if hasattr(vevent, 'uid'):
                 uid = vevent.uid.value
             data = {
+                # CO:
+                # Some custom cards access 'id' and 'summary' of event object
+                # because google calendar integration / google calendar event
+                # API has it.
+                #
+                # CO: Here a "uid" is given (from caldav spec),
+                # but it is "id" in google calendar integration
+                #                                / google calendar event API.
                 "uid": uid,
+                "id": uid,
```

Comment:
We should follow the vcard spec, not Google or whatever custom cards expect. Custom cards should adopt to our standard, we not to theirs.

Reply:
@balloob: 
vcard is IMHO for contacts and caldav uses iCalendar (https://tools.ietf.org/html/rfc5545) entries.

But in general totally agreed that custom_cards should follow home assistant and not the other way around. 
However:
* Currently the official google calendar component just forwards what it gets from google (https://developers.google.com/calendar/v3/reference/events), 
* The official webdav calendar component follows iCalendar, but not 100%, title versus summary.

Compare GoogleCalendarData::update/GoogleCalendarData::async_get_events with WebDavCalendarData::update/WebDavCalendarData::async_get_events

Reply:
Sorry, you're right, I meant RFC5545.

Reply:
@oechslein can you adjust the PR according to comments or do you need more input?

Reply:
The comment indicates that I need to change the google component and not the caldav component. That will be a breaking change for custom components that show google calendar.
But in general I think I know what to do, just need some time to do it.